ospfd/ospf_packet.c

```diff
@@ -2091,6 +2091,24 @@ static void ospf_ls_upd(struct ospf *ospf, struct ip *iph,
 				continue;
 		}
 
+		/* Added to prevent the router from accepting old LSAs that were
+		sent by it
+		before, were not properly removed (e.g. due to crash) but which
+		have become invalid */
```

Comment:
I don't think this is a proper solution to the problem. *ospfd* already has mechanisms to handle the receipt of self-originated LSAs once the daemon is restarted. Those LSAs are either reoriginated or flushed depending whether they are still valid or not. You can look at `ospf_process_self_originated_lsa()` and `ospf_abr_task_timer()` for more details. I think the bug you reported should be fixed within the scope of those functions.

I've commented on #13732 asking for more information so that I can understand the problem better. I couldn't reproduce the reported issue locally.